Fabric.Identity.API/scripts/tests/Remove-IdentityEncryptionCertificate.tests.ps1

```diff
@@ -0,0 +1,100 @@
+                    Name = "FriendlyName"
+                    value = "Fabric Identity Signing Encryption Certificate"
+                    Force = $true
+                }
+                $testCert | Add-Member @friendlyNameProperty -ErrorAction 0
+                $thumbPrint = "non-existing-thumbprint"
+                Mock -ModuleName Install-Identity-Utilities -CommandName Get-Certificate -MockWith { return $testCert }
+                Mock -ModuleName Install-Identity-Utilities -CommandName Write-DosMessage -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName Remove-Item -MockWith {}
+    
+                # Act
+                Remove-IdentityEncryptionCertificate -encryptionCertificateThumbprint $thumbPrint
+    
+                # Assert
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Write-DosMessage -ParameterFilter { $Level -and $Level -eq "Information" -and $Message.StartsWith("Removing Identity encryption certificate") } -Times 1 -Exactly -Scope It
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Write-DosMessage -Times 1 -Exactly -Scope It
+            }
+        }
+    }
+
+    Context 'Should not remove Certificate' {
+        InModuleScope Install-Identity-Utilities {
+            It 'Should not error and should not attempt to remove when certificate is not found' {
+                # Arrange
+                $thumbPrint = "non-existing-thumbprint"
+                Mock -ModuleName Install-Identity-Utilities -CommandName Write-DosMessage -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName Remove-Item -MockWith {}
+
+                # Act
+                Remove-IdentityEncryptionCertificate -encryptionCertificateThumbprint $thumbPrint
+
+                # Assert
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Write-DosMessage -Times 1 -Exactly -Scope It
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Write-DosMessage -ParameterFilter { $Level -and $Level -eq "Information" -and $Message.StartsWith("Certificate with thumbprint") } -Times 1 -Exactly -Scope It
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Remove-Item -Times 0 -Exactly -Scope It
+            }
+        }
+
+        InModuleScope Install-Identity-Utilities {
+            It 'Should not attempt to remove when certificate is not the identity encryption certificate' {
+                # Arrange
+                $testCert = New-Object -TypeName System.Security.Cryptography.X509Certificates.x509Certificate2
+                $friendlyNameProperty = @{
+                    MemberType = [System.Management.Automation.PSMemberTypes]::NoteProperty
+                    Name = "FriendlyName"
+                    value = "Some Friendly Name"
```

Comment:
So the value in the friendlyNameProperty would have to be "Fabric Identity Signing Encryption Certificate" for it to be removed?

Reply:
Ya, it defaults to "Fabric Identity Signing Encryption Certificate" to look for, but that can be passed in to be something else.

The idea was to not accidently remove the SSL certificate, or any certificate that we did not create ourselves.